### PR TITLE
Add QLinearMul operator

### PR DESCRIPTION
**Pull request by gyulaz-htec**
[QLinearMul](https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#com.microsoft.QLinearMul) operator is required for the [DenseNet-int8](https://github.com/onnx/models/blob/main/vision/classification/densenet-121/model/densenet-12-int8.onnx) onnx zoo model.

Fixes: https://github.com/migraphx-benchmark/AMDMIGraphX/issues/150

### CODE REVIEW

**Comment by pfultz2** [comment]
This should reuse the `parse_qlinearadd` class, but it should be renamed to `parse_qlinearbinary`, you can use `opd.op_name` to get the operator name:

```cpp
struct parse_qlinearbinary : op_parser<parse_qlinearbinary>
{
    std::vector<op_desc> operators() const { return {{"QLinearAdd", "add"}, {"QLinearMul", "mul"}}; }

    void check_inputs(...) const
    {
        ...
    }

    instruction_ref parse(const op_desc& opd,
                          const onnx_parser& /*parser*/,
                          const onnx_parser::node_info& info,
                          const std::vector<instruction_ref>& args) const
    {
        check_inputs(args);

        // A
        const auto& in_a         = args[0];
        const auto& in_scale_a   = args[1];
        const auto& in_zero_pt_a = args[2];

        auto dquant_a = bcast_qdq_instr("dequantizelinear", in_a, in_scale_a, in_zero_pt_a, info);

        // B
        const auto& in_b         = args[3];
        const auto& in_scale_b   = args[4];
        const auto& in_zero_pt_b = args[5];
        auto dquant_b = bcast_qdq_instr("dequantizelinear", in_b, in_scale_b, in_zero_pt_b, info);

        // C = A + B
        auto out_c = info.add_common_op(opd.op_name, dquant_a, dquant_b);

        const auto& in_scale_c = args[6];

        // zero_pt for C is supplied as the last optional argument..
        if(args.size() == 8)
            return (bcast_qdq_instr("quantizelinear", out_c, in_scale_c, args[7], info));

        // if no zero_pt: just broadcast the scale..
        auto bcast_scale_c = bcast_scalar_instr(out_c->get_shape(), in_scale_c, info);
        return (info.add_instruction(migraphx::make_op("quantizelinear"), out_c, bcast_scale_c));
    }
};
```

**Review by lakhinderwalia — COMMENTED** [review]
Could you please help me understand how you could use the exact test data-set from qlinearadd** tests to verify for qlinearmul_test & qlinear_bcast_test. The compute results for Add and Mul operations should be (very) different. Thanks.

**Comment by gyulaz-htec** [comment]
@lakhinderwalia I've lowered the scales for [A](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/2430/files#diff-78796ec3c274d2b01cb1d0ad414dd139fb26b4cbb5c00ae5789fbe3f23b9f3c1R6014) and [B](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/2430/files#diff-78796ec3c274d2b01cb1d0ad414dd139fb26b4cbb5c00ae5789fbe3f23b9f3c1R6018) from `0.05` to `0.005` and increased the scale for [C](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/2430/files#diff-78796ec3c274d2b01cb1d0ad414dd139fb26b4cbb5c00ae5789fbe3f23b9f3c1R6021) compared to QLinearAdd.
With these changes the output of `((A - A_zero_point) * (B - B_zero_point)) * (A_scale * B_scale)/C_scale` is below `0.5`, so adding it to to [C_zero_point](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/2430/files#diff-78796ec3c274d2b01cb1d0ad414dd139fb26b4cbb5c00ae5789fbe3f23b9f3c1R6022) will always output the value of `C_zero_point`

**Comment by gyulaz-htec** [comment]
@pfultz2 @lakhinderwalia I've updated the PR with the proposed `parse_qlinearadd` struct. Also if you want me to change the test to use more diverse results/scale values/zero points, let me know and I will update it.

**Comment by lakhinderwalia** [comment]
> @lakhinderwalia I've lowered the scales for [A](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/2430/files#diff-78796ec3c274d2b01cb1d0ad414dd139fb26b4cbb5c00ae5789fbe3f23b9f3c1R6014) and [B](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/2430/files#diff-78796ec3c274d2b01cb1d0ad414dd139fb26b4cbb5c00ae5789fbe3f23b9f3c1R6018) from `0.05` to `0.005` and increased the scale for [C](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/2430/files#diff-78796ec3c274d2b01cb1d0ad414dd139fb26b4cbb5c00ae5789fbe3f23b9f3c1R6021) compared to QLinearAdd. With these changes the output of `((A - A_zero_point) * (B - B_zero_point)) * (A_scale * B_scale)/C_scale` is below `0.5`, so adding it to to [C_zero_point](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/2430/files#diff-78796ec3c274d2b01cb1d0ad414dd139fb26b4cbb5c00ae5789fbe3f23b9f3c1R6022) will always output the value of `C_zero_point`

Please note, all the output values that you are testing/verifying are zeros. This test-vector originally designed for _Add_ isn't right for _Multiplication_. It is likely that your scale is so small that it is all effectively converting to zeros -- and the test passes. Thanks.

**Comment by gyulaz-htec** [comment]
@lakhinderwalia I've changed the scale values so the in between values are not zeros, also updated the results according to that.

File: test/onnx/gen_onnx.py [hunk 1]
```diff
@@ -6008,6 +6008,60 @@ def qlinearmatmul_3D_test():
...
 
 
+@onnx_test()
+def qlinearmul_test():
+    a = helper.make_tensor_value_info('A', TensorProto.UINT8, [64])
+    sc_a = helper.make_tensor('A_scale', TensorProto.FLOAT, [], [0.05])
+    zero_pt_a = helper.make_tensor('A_zero_point', TensorProto.UINT8, [], [0])
+
+    b = helper.make_tensor_value_info('B', TensorProto.UINT8, [64])
+    sc_b = helper.make_tensor('B_scale', TensorProto.FLOAT, [], [-0.05])
```
Comment (lakhinderwalia) [inline test/onnx/gen_onnx.py:6018]: Quantization scale is always a positive number.

Comment (gyulaz-htec) [reply test/onnx/gen_onnx.py]: Changed it to positive number, and the test results according to that.

**Review by lakhinderwalia — APPROVED** [review]
Thank you.

**Review by pfultz2 — APPROVED** [review]
